tests/comments/test_async_api_org_terms.py

```diff
@@ -0,0 +1,320 @@
+    url = reverse(
+        'comments_async-detail',
+        kwargs={
+            'pk': comment.pk,
+            'content_type': comment.content_type.pk,
+            'object_pk': comment.object_pk
+        })
+    data = {
+        'comment': 'comment comment comment',
+        'agreed_terms_of_use': True
+    }
+
+    with active_phase(comment.content_object.module, AskPhase):
+        response = apiclient.patch(url, data, format='json')
+        assert response.status_code == status.HTTP_200_OK
+        assert response.data['comment'] == 'comment comment comment'
+
+    terms = OrganisationTermsOfUse.objects.all()
+    assert len(terms) == 0
+
+
+@pytest.mark.django_db
+@override_settings(
+    A4_USE_ORGANISATION_TERMS_OF_USE=True
+)
+def test_comment_save_without_agreement_forbidden(
+        user, apiclient, question_ct, question):
+    apiclient.force_authenticate(user=user)
+    url = reverse(
+        'comments_async-list',
+        kwargs={'content_type': question_ct.pk,
+                'object_pk': question.pk})
+    data = {
+        'comment': 'comment comment',
+        'agreed_terms_of_use': False
+    }
+    with active_phase(question.module, AskPhase):
+        response = apiclient.post(url, data, format='json')
+
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
+
+
+@pytest.mark.django_db
+@override_settings(
+    A4_USE_ORGANISATION_TERMS_OF_USE=True
+)
+def test_comment_save_without_agreement_data_forbidden(
+        user, apiclient, question_ct, question):
+    apiclient.force_authenticate(user=user)
+    url = reverse(
+        'comments_async-list',
+        kwargs={'content_type': question_ct.pk,
+                'object_pk': question.pk})
+    data = {
+        'comment': 'comment comment'
+    }
+    with active_phase(question.module, AskPhase):
+        response = apiclient.post(url, data, format='json')
+
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
+
+
+@pytest.mark.django_db
+@override_settings(
+    A4_USE_ORGANISATION_TERMS_OF_USE=True
+)
+def test_comment_save_with_already_agreed(
+        user, apiclient, question_ct, question,
+        organisation_terms_of_use_factory):
+    organisation_terms_of_use_factory(
+        user=user,
+        organisation=question.module.project.organisation,
+        has_agreed=True,
+    )
+    apiclient.force_authenticate(user=user)
+    url = reverse(
+        'comments_async-list',
+        kwargs={'content_type': question_ct.pk,
+                'object_pk': question.pk})
+    data = {
+        'comment': 'comment comment',
+    }
+    with active_phase(question.module, AskPhase):
+        response = apiclient.post(url, data, format='json')
+
+    assert response.status_code == status.HTTP_201_CREATED
+    assert response.data['comment'] == 'comment comment'
+
+
+@pytest.mark.django_db
+@override_settings(
+    A4_USE_ORGANISATION_TERMS_OF_USE=True
+)
+def test_agreement_fields_not_implemented(
+        apiclient, question_ct,
```

Comment:
Can you explain to me what this test does? Somehow dont get it, sorry..

Reply:
It tests the new line in the API that throws the otImplementedError (lines 226 and 227 of adhocracy4/comments_async/api.py)
I think, I should redo the test naming. It's really bad and does not make a lot of sense.